Pylint: Fix `consider-using-with` issues.

## 💡 Description
Update the codebase to use `with` context manager for opening files. `pylint` report is attached (pylint . --disable=all --enable=consider-using-with`):
```
************* Module pds.naif_pds4_bundler.classes.label.pds4_inventory
pds/naif_pds4_bundler/classes/label/pds4_inventory.py:64:12: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
************* Module pds.naif_pds4_bundler.utils.files
pds/naif_pds4_bundler/utils/files.py:605:20: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
pds/naif_pds4_bundler/utils/files.py:855:19: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
pds/naif_pds4_bundler/utils/files.py:863:19: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
```

--- a/src/pds/naif_pds4_bundler/classes/label/pds4_inventory.py
+++ b/src/pds/naif_pds4_bundler/classes/label/pds4_inventory.py
@@ -61,9 +61,8 @@ class InventoryPDS4Label(PDSLabel):
         #
         # Count number of lines in the inventory file
         #
-        f = open(self.product.path, encoding='utf-8')
-        self.N_RECORDS = str(len(f.readlines()))
-        f.close()
+        with open(self.product.path, 'r', encoding='utf-8') as f:
+            self.N_RECORDS = str(len(f.readlines()))
 
         self.name = collection.name.split(".")[0] + ".xml"
         self.write_label()

--- a/src/pds/naif_pds4_bundler/utils/files.py
+++ b/src/pds/naif_pds4_bundler/utils/files.py
@@ -12,6 +12,7 @@ import re
 import shutil
 import stat
 import sys
+from pathlib import Path
 from typing import Optional
 
 import spiceypy
@@ -602,15 +603,10 @@ def compare_files(fromfile, tofile, dest_dir, display):
             fromlines, tolines, fromfile, tofile, context=False, numlines=False
         )
 
-        diff_html = open(
-            dest_dir + f"/diff_"
-            f"{fromfile.split(os.sep)[-1].replace('.', '_')}_"
-            f"{tofile.split(os.sep)[-1].replace('.', '_')}"
-            f".html",
-            "w", encoding='utf-8'
-        )
-        diff_html.writelines(diff)
-        diff_html.close()
+        with open(f'{dest_dir}/diff_{Path(fromfile).name.replace(".", "_")}_'
+                  f'{Path(tofile).name.replace(".", "_")}.html',
+                  'w', encoding='utf-8') as diff_html:
+            diff_html.writelines(diff)
 
     return True
 
@@ -852,19 +848,17 @@ def replace_string_in_file(file, old_string, new_string, setup):
     :type new_string: str
     :param setup: NPB run Setup
     """
-    reading_file = open(file, "r", encoding='utf-8')
+    with open(file, 'r', encoding='utf-8') as f:
 
-    new_file_content = ""
-    for line in reading_file:
-        new_line = line.replace(old_string, new_string)
-        new_file_content += add_carriage_return(new_line, setup.eol_pds3, setup)
-    reading_file.close()
+        new_file_content = ''
+        for line in f:
+            new_line = line.replace(old_string, new_string)
+            new_file_content += add_carriage_return(new_line, setup.eol_pds3, setup)
 
-    writing_file = open("temp.file", "w", encoding='utf-8')
-    writing_file.write(new_file_content)
-    writing_file.close()
+    with open('temp.file', 'w', encoding='utf-8') as writing_file:
+        writing_file.write(new_file_content)
 
-    shutil.move("temp.file", file)
+    shutil.move('temp.file', file)
 
 
 def format_multiple_values(value):

